### PR TITLE
Ticket 8 add comment

### PR DESCRIPTION
**Pull request by ElenaCKay**
I have added a sign in page which renders all users and allows the user to pick one to 'sign in'. This is then used when posting a comment as it displays the username. I have also added in a carousel feature on the main reviews page.

### CODE REVIEW

File: src/api.js [hunk 1]
```diff
@@ -27,3 +27,16 @@ export const patchReviewVotes = (id, incVotes) => {
         return response.data.review.votes;
     });
 };
+
+export const postComment = (id, { username, body }) => {
+    console.log(id, username, body);
```
Comment (B7Bprog) [inline src/api.js]: Please remove this log.

**Review by B7Bprog — CHANGES_REQUESTED** [review]
Hey Elena, there are a few points to check and correct:

- When not logged in, a user shouldn't be able to type and click to send a post. Also should be notified that they need to log in first and the post wasn't successful.
- When the connection is slow, the user should be notified that the comment is being posted or the form should be disabled to avoid multiple attempts.
- When the internet is down and the post is unsuccessful, there should be a message displayed.
- The input field should be a 'textarea' to allow multiline comments. This should be disabled when not logged in.
- Should not post when input field is empty as it currently results in an error. 

Please try to refactor your code and send us another PR with it.

File: src/App.js [hunk 2]
```diff
@@ -5,18 +5,27 @@ import Reviews from "./components/reviews";
...
+                    path="/signIn"
+                    element={<SignInPage user={user} setUser={setUser} signedIn={signedIn} setSignedIn={setSignedIn} />}
+                />
                 <Route path="/" element={<Reviews />} />
-                <Route path="/reviews" element={<Reviews />} />
-                <Route path="/reviews/:review_id" element={<ReviewCard />} />
-                <Route path="/reviews/:review_id" element={<Comments />} />
+                <Route path="/reviews" element={<Reviews signedIn={signedIn} user={user}/>} />
+                <Route path="/reviews/:review_id" element={<ReviewCard user={user} signedIn={signedIn} />} />
+                <Route path="/reviews/:review_id" element={<Comments user={user} signedIn={signedIn}/>} />
```
Comment (HeatherEver) [inline src/App.js]: there should not be code commented out in 'production code' - remember to keep things tidy! (line 29)

File: src/components/reviewItem.jsx [hunk 3]
```diff
@@ -0,0 +1,44 @@
+import { Link } from "react-router-dom";
+import Carousel from "react-bootstrap/Carousel";
+
+const ReviewItem = ({ reviewsData }) => {
+    return (
```
Comment (HeatherEver) [inline src/components/reviewItem.jsx]: more commented out code below - remember to keep things tidy!

File: src/components/reviewsHeader.jsx [hunk 4]
```diff
@@ -0,0 +1,26 @@
+import { Link } from "react-router-dom";
+
+const ReviewsHeader = ({ signedIn, user }) => {
+    console.log(signedIn);
```
Comment (HeatherEver) [inline src/components/reviewsHeader.jsx]: remove console logs

**Review by HeatherEver — COMMENTED** [review]
It's looking really good but there's still just a couple of things to look at: 
- comment box needs to be a text area so the user can post a long comment and still be able to see what they've typed
- i can post an empty space " " which isn't too bad and something which can be looked at later (think about `.trim()`) 
- there's a warning in your console regarding a list element not having a key, make sure you're fixing your warning's before sending pull requests 😊

**Review by Pipe-Smoking-Rabbit — CHANGES_REQUESTED** [review]
Hey Elena! It's looking fantastic, I do have some suggestions for changes on user feedback:

- When posting a comment successfully you are giving some great user feedback in the form of the text that reads "Adding your comment" 🏆  The only thing I noticed was that the entire comment section temporarily disappears, if you could make it so the comments remain on the screen but only the text input box disappears I think that would be spot on 👍 
- The other thing is that when posting in offline mode the message "Adding your comment" just hangs on the screen forever, it would be really nice to get a catch in there to render a different message informing the user of when their request has failed 

The first suggestion is more of an optional "would be nice", the second is the more important of the two. Once you've got that going hit us up with one last PR and we can get it ticked off! 😄

File: src/components/addComment.jsx [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+        setNewCommentLoading(true);
+        postComment(review_id, commentToAdd)
+            .then((response) => {
+                setNewCommentLoading(false);
+                setComments((currComments) => {
+                    return [response, ...currComments];
+                });
+            })
+            .catch((err) => {
+                console.log("im in the catch");
```
Comment (Hard-Luck) [inline src/components/addComment.jsx]: need to remove console.logs and comments

File: src/components/addComment.jsx [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+        return (
+            <div>
+                <h3>Adding your comment...</h3>
+            </div>
+        );
+    }
+
+    return (
+        <form onSubmit={submitComment}>
+            <label>{commentFailed.toString()}</label>
```
Comment (Hard-Luck) [inline src/components/addComment.jsx]: image
This is rendering on the page as "false"

**Review by Hard-Luck — COMMENTED** [review]
Just a couple of things need sorting out, see my comments :) then you can merge

File: src/api.js [hunk 7]
```diff
@@ -4,8 +4,10 @@ const gamesAPI = axios.create({
     baseURL: "https://games-backend-project.onrender.com",
 });
 
-export const fetchReviews = () => {
-    return gamesAPI.get(`/api/reviews`).then((response) => {
+export const fetchReviews = (category) => {
+    //console.log(category);
+    return gamesAPI.get(`/api/reviews`, { params: { category: category } }).then((response) => {
+        console.log(category)
```
Comment (Poonam-raj) [inline src/api.js]: Please remove all console logs before thinking about merging this code 🙏🏾

File: src/api.js [hunk 8]
```diff
@@ -4,8 +4,10 @@ const gamesAPI = axios.create({
     baseURL: "https://games-backend-project.onrender.com",
 });
 
-export const fetchReviews = () => {
-    return gamesAPI.get(`/api/reviews`).then((response) => {
+export const fetchReviews = (category) => {
+    //console.log(category);
```
Comment (Poonam-raj) [inline src/api.js]: Let's delete all commented out bits of code before merging

**Review by Poonam-raj — CHANGES_REQUESTED** [review]
I have screenshotted one requirement from the ticket
image

Please let the URL reflect what category I'm filtering the reviews by so I can share with a friend. Submit another PR once that's done and then i think you're good to go

**Review by Poonam-raj — APPROVED** [review]
Awesome work - thanks for amending!

One thing I noticed was that when i refresh the drop-down resets despite being on the same filtered page:
image


**Brill work - feel free to merge this branch! 🎉 **

**Pull request merged**